diff.py

```diff
@@ -1422,12 +1582,15 @@ class ArchSettings:
     re_sprel: Pattern[str]
     re_large_imm: Pattern[str]
     re_imm: Pattern[str]
+    re_reloc: Pattern[str]
     branch_instructions: Set[str]
     instructions_with_address_immediates: Set[str]
     forbidden: Set[str] = field(default_factory=lambda: set(string.ascii_letters + "_"))
     arch_flags: List[str] = field(default_factory=list)
     branch_likely_instructions: Set[str] = field(default_factory=set)
     difference_normalizer: Type[DifferenceNormalizer] = DifferenceNormalizer
+    reloc_proc: Type[RelocationProcessor] = RelocationProcessor
+    post_proc: Type[PostProcessor] = PostProcessor
```

Comment:
Can you fold all of these three processor classes into a single class? They fill more or less the same purpose of acting as arch hooks for the asm parsing process.

(Alternatively reoc_proc and post_proc could be made into functions, since the classes are stateless)

Reply:
Done, folded them into a single class.